app/models/merchant.rb

```diff
@@ -1,5 +1,18 @@
 class Merchant < ApplicationRecord
   has_many :items
+  has_many :invoice_items, through: :items
+  has_many :invoices, through: :invoice_items
+  has_many :transactions, through: :invoices
+  has_many :customers, through: :invoices
 
   validates :name, presence: true
+
+  def top_five_customers
+    self.transactions.joins(invoice: :customer)
+      .where("transactions.result = ?", "0")
+      .select("customers.id, CONCAT(customers.first_name,' ', customers.last_name) as full_name, COUNT(transactions) as successful_transactions")
```

Comment:
Rather than making an alias under the `full_name`, it'd be worth considering switching it to a customer model method. Finding where it comes from would be harder to find as an alias later on.